webapp/src/components/MapShowLocations/index.js

```diff
@@ -31,50 +158,136 @@ function MapShowLocations({ location, ...props }) {
     skip: true
   })
 
-  useEffect(() => {
-    const handleEvent = async (map) => {
-      const { lng, lat } = map.getCenter()
-
-      const { data } = await getNearbyLocations({
-        distance,
-        point: {
-          type: 'Point',
-          coordinates: [lng, lat]
+  const loadPointData = (name, type, coordinates) => {
+    customData.features.push(
+      {
+        'type': 'Feature',
+        'properties': {
+          'title': name,
+          'type': type
+        },
+        'geometry': {
+          'coordinates': [coordinates[0], coordinates[1]],
+          'type': 'Point'
+        }
+      }
+    );
+  }
+
+  const claerPointsData = () => {
+    customData.features = []
+    markerList = []
+  }
+
+  const removeMarkers = () => {
+    for (var i = 0; i <= markerList.length - 1; i++) {
+      markerList[i].remove();
+    }
+  }
+
+  const loadMarkersEvent = async () => {
+    removeMarkers()
+    claerPointsData()
+
+    const { lng, lat } = map.getCenter()
+
+    const { data } = await getNearbyLocations({
+      distance: searchDistance,
+      point: {
+        type: 'Point',
+        coordinates: [lng, lat]
+      }
+    })
+
+    data &&
+      data.locations &&
+      data.locations.forEach((location) => {
+        const {
+          id,
+          account,
+          type,
+          geolocation: { coordinates },
+          info
+        } = location
+
+        loadPointData(info.name, type, coordinates)
+
+        const markerNode = document.createElement('div')
+        ReactDOM.render(<MapMarker type={type} />, markerNode)
+
+        const popupNode = document.createElement('div')
+        ReactDOM.render(
+          <MapPopup id={id} info={info} account={account} />,
+          popupNode
+        )
+
+        var markertemp = new mapboxgl.Marker(markerNode)
```

Comment:
avoid to use var, you can use const or let if you need to modified the variable value